.github/workflows/build_and_test.yaml

```diff
@@ -0,0 +1,66 @@
+# Please reference work here https://github.com/EVerest/everest-core/tree/main/.github/workflows
+# TODO: modify to reuse the above workflow to DRY up CI.
+
+name: Build and test libtimer
+on: 
+  pull_request: {}
+  workflow_dispatch:
+    inputs:
```

Comment:
In https://github.com/EVerest/liblog/pull/16 we are running this on every push and every pull request. Here, we are only running on every PR. I think we should standardize on one of them. I think that on every PR is likely enough, although I don't think it hurts to run one final time on push.

If this runs only on PRs, though, I wonder how you tested it in
https://github.com/US-JOET/libtimer/actions

Did you originally have it on push but then commented it out?

Reply:
Based on https://github.com/EVerest/libfsm/pull/8#issuecomment-1852127925, it looks like the current convention is to enable on every push only for testing, and only on PRs thereafter. Since this is likely to be revisited when switching to reusable workflows, I will approve this. But I do think that this should be documented as the best practice going forward for future CI/CD testers to use.

Reusable workflows don't really control the `on`